src/app/auth/components/signup/signup.component.ts

```diff
@@ -107,17 +123,87 @@ export class SignupComponent {
       name: [name || '', Validators.required],
       password: [
         '',
-        [Validators.required, Validators.minLength(MIN_PASSWORD_LENGTH)],
+        [
+          Validators.required,
+          Validators.minLength(MIN_PASSWORD_LENGTH),
+          ...(this.enabledPasswordCheckStrength
+            ? [this.passwordStrengthValidator()]
+            : []),
+        ],
       ],
     });
-
     const confirmPasswordControl = new UntypedFormControl('', [
       Validators.required,
       matchControlValidator(this.signupForm.controls['password']),
     ]);
     this.signupForm.addControl('confirm', confirmPasswordControl);
   }
 
+  ngOnInit(): void {
+    this.passwordSubscription = this.signupForm.controls[
+      'password'
+    ].valueChanges.subscribe((password) => {
+      if (this.featureFlagService.isEnabled('passwordStrngth')) {
```

Comment:
```suggestion
      if (this.featureFlagService.isEnabled('password-strength')) {
```

There's a typo in this flag name. We might also want to use "kebab-case" for feature flags. I think it might be the most accessible to people on the experience team, though we don't have a defined standard yet so it might be fine to leave it like this.

As a small note, I like naming a component's instance of the FeatureFlagService just `feature` so that the code can read nicely: `this.feature.isEnabled('feature-name')`, but it doesn't really matter what it's named and it's easy to refactor later.